tests/torch/test_graph_building.py

```diff
@@ -579,7 +580,30 @@ def test_integer_path_marking():
     nncf_graph = builder.build_graph(ModelWithIntegerPaths(), input_infos=input_infos)
     edges = list(nncf_graph.get_all_edges())
     num_integer_edges = sum([1 for edge in edges if edge.dtype is Dtype.INTEGER])
-    assert num_integer_edges == 2  # cat -> __floordiv__ and __floordiv__ -> __getitem__
+    # cat -> __floordiv__,  __floordiv__ -> __getitem__0 (to get single_idx),
+    # __getitem__0 -> __getitem__1 (first indexing by tensor), __getitem__0 -> __getitem__2 (second indexing by tensor)
+    assert num_integer_edges == 4
+
+
+def test_torch_tensor_getitem_behavior(mocker):
+    x = torch.ones((10, 4, 4, 4))
+    indexes = torch.LongTensor([0, 1, 2])
+    mock_tensor_meta = mocker.stub
+    traced_x = TracedTensor.from_torch_tensor(torch.ones((10, 4, 4, 4)), mock_tensor_meta)
+    traced_indexes = TracedTensor.from_torch_tensor(torch.LongTensor([0, 1, 2]), mock_tensor_meta)
+    SHAPE_1 = [3, 4, 4, 4]
+    SHAPE_2 = [3, 4, 4]
+    assert list(x[indexes].shape) == SHAPE_1
```

Comment:
Added tests from ticket 82065 for `torch.Tensor.__getitem__` behavior, which is fixed since torch 1.11